### PR TITLE
Update repo to isort v5

### LINKED ISSUE #6681: BUILD upgrade to isort v5
The contributing guide says

> Right now, we need an unreleased version of isort to avoid a bug:

However, it looks like the linked PR has been incorporated into the latest version of isort. I could submit a PR to address this if it would be welcome

### PR DESCRIPTION
**Pull request by sgugger**
Since isort now works properly with black, we can use the latest version. It also comes with new functionality (hence the lot of diff) mainly:
- it can deal with the __init__
- it can deal with import in if/else blocks.

This will fix #6681

Also v5 does not use the recursive flag anymore, so I removed it from the make style and make quality commands. For users having an old isort version, this will result in make style/make quality having no impact.

It's very likely that users having suggested a PR will need to rebase after this is merged, update isort then run a new `make style`. We can also assist by force-pushing on their branches.

Commented on the changes I made manually, all the others come from the new version of isort.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -235,8 +235,7 @@ jobs:
                       - v0.3-code_quality-{{ checksum "setup.py" }}
                       - v0.3-{{ checksum "setup.py" }}
             - run: pip install --upgrade pip
-            # we need a version of isort with https://github.com/timothycrosley/isort/pull/1000
-            - run: pip install git+git://github.com/timothycrosley/isort.git@e63ae06ec7d70b06df9e528357650281a3d3ec22#egg=isort
+            - run: pip install isort
```
Comment (sgugger) [inline .circleci/config.yml:238]: Update the CI to use the new isort.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -134,12 +134,6 @@ Follow these steps to start contributing:
    it with `pip uninstall transformers` before reinstalling it in editable
    mode with the `-e` flag.)
 
-   Right now, we need an unreleased version of `isort` to avoid a
-   [bug](https://github.com/timothycrosley/isort/pull/1000):
-
-   ```bash
-   $ pip install -U git+git://github.com/timothycrosley/isort.git@e63ae06ec7d70b06df9e528357650281a3d3ec22#egg=isort
-   ```
```
Comment (sgugger) [inline CONTRIBUTING.md:142]: No need for this section in the CONTRIBUTING since the version installed should just work.

File: Makefile [hunk 3]
```diff
@@ -4,15 +4,15 @@
 
 quality:
 	black --check --line-length 119 --target-version py35 examples templates tests src utils
-	isort --check-only --recursive examples templates tests src utils
+	isort --check-only examples templates tests src utils
```
Comment (sgugger) [inline Makefile:7]: As mentioned, --recursive flag will raise an error in isort v5.

File: setup.cfg [hunk 4]
```diff
@@ -1,4 +1,5 @@
 [isort]
+default_section = FIRSTPARTY
```
Comment (sgugger) [inline setup.cfg:2]: This is to ensure isort behaves like the old version (to minimize the diff, removing this changes a new 48 files).

File: setup.py [hunk 5]
```diff
@@ -91,12 +91,7 @@
 extras["testing"] = ["pytest", "pytest-xdist", "timeout-decorator", "psutil"]
 # sphinx-rtd-theme==0.5.0 introduced big changes in the style.
 extras["docs"] = ["recommonmark", "sphinx", "sphinx-markdown-tables", "sphinx-rtd-theme==0.4.3", "sphinx-copybutton"]
-extras["quality"] = [
-    "black",
-    # "isort",
-    "isort @ git+git://github.com/timothycrosley/isort.git@e63ae06ec7d70b06df9e528357650281a3d3ec22#egg=isort",
-    "flake8",
-]
+extras["quality"] = ["black", "isort >= 5", "flake8"]
```
Comment (sgugger) [inline setup.py:94]: Pins isort to >=5

File: src/transformers/benchmark/benchmark.py [hunk 6]
```diff
@@ -22,14 +22,9 @@
 import timeit
 from typing import Callable, Optional
 
-from transformers import (
```
Comment (sgugger) [inline src/transformers/benchmark/benchmark.py:25]: Use relative imports.

File: src/transformers/benchmark/benchmark_tf.py [hunk 7]
```diff
@@ -24,14 +24,9 @@
 from functools import wraps
 from typing import Callable, Optional
 
-from transformers import (
```
Comment (sgugger) [inline src/transformers/benchmark/benchmark_tf.py:27]: Use relative imports

**Review by patrickvonplaten — APPROVED** [review]
Awesome - thanks!

**Review by julien-c — APPROVED** [review]

**Review by sshleifer — APPROVED** [review]
Big moves!

**Review by LysandreJik — APPROVED** [review]
Great, LGTM! Thanks @sgugger

**Pull request merged**